icu4c/source/tools/icu-file-utf8-check.py

```diff
@@ -84,23 +94,30 @@ def main(argv):
     if args:
         print("unexpected command line argument")
         usage()
-        sys.exit()
-
-    output = runCommand("git ls-files ");
-    file_list = output.splitlines()
-
-    source_file_re = re.compile(".*((?:\\.c$)|(?:\\.cpp$)|(?:\\.h$)|(?:\\.java$))")
-    
-    for f in file_list:
-        if os.path.isdir(f):
-            print("Skipping dir " + f)
-            continue
-        if not os.path.isfile(f):
-            print("Repository file not in working copy: " + f)
-            continue;
-
-        source_file = source_file_re.match(f)
-        check_file(f, source_file)
+        sys.exit(2)
+
+    git_cmd = "git ls-files DIR"
+    for dir in icu_directories_to_be_scanned:
+        print('Scanning ' + dir)
+        output = runCommand(git_cmd.replace("DIR", dir))
+        file_list = output.splitlines()
+
+        source_file_re = re.compile(".*((?:\\.c$)|(?:\\.cpp$)|(?:\\.h$)|(?:\\.java$))")
```

Comment:
I wonder if this regex could be moved outside of the for loop.
(Not that perf is really that big of a concern here though...)

Reply:
Valid point -done.